examples/python/async_streaming/client.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2020 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import threading
+from typing import Iterator
+from concurrent.futures import ThreadPoolExecutor
+
+import grpc
+
+import phone_pb2
+import phone_pb2_grpc
+
+
+class CallMaker:
+
+    def __init__(self, executor: ThreadPoolExecutor, channel: grpc.Channel,
+                 phone_number: str) -> None:
+        self._executor = executor
+        self._channel = channel
+        self._stub = phone_pb2_grpc.PhoneStub(self._channel)
+        self._phone_number = phone_number
+        self._session_id = None
+        self._audio_session_link = None
+        self._call_state = None
+        self._peer_responded = threading.Event()
+        self._call_finished = threading.Event()
+        self._consumer_future = None
+
+    def _response_watcher(
+            self,
+            response_iterator: Iterator[phone_pb2.StreamCallResponse]) -> None:
+        try:
+            for response in response_iterator:
+                if response.HasField("call_info"):
+                    self._on_call_info(response.call_info)
+                elif response.HasField("call_state"):
+                    self._on_call_state(response.call_state.state)
```

Comment:
Nit: Probably a good idea to explicitly handle the case that neither of these fields is non-default.

Reply:
Added an extra else to raise if none of above fields present.